File: src/test/java/two_sum/TwoSumTest.java
package two_sum;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertArrayEquals;

class TwoSumTest {

    @Test
    void test() {
        TwoSum twoSum = new TwoSum();
        int[] nums = {2, 7, 11, 15};
        int target = 9;

        int[] output1 = twoSum.twoSum1(nums, target);
        assertArrayEquals(new int[]{0, 1}, output1);

        int[] output2 = twoSum.twoSum2(nums, target);
        assertArrayEquals(new int[]{0, 1}, output2);
    }
}

File: src/test/kotlin/array/TwoSumTest.kt
package array

import org.assertj.core.api.Assertions
import org.junit.jupiter.api.Test

class TwoSumTest {
    @Test
    fun `should return indices of array whose sum is equal to target`() {
        Assertions.assertThat(TwoSum().twoSum(intArrayOf(-1,-2,-3,-4,-5), -8)).containsExactlyInAnyOrder(2, 4)
        Assertions.assertThat(TwoSum().twoSum(intArrayOf(11, 15, 2, 7), 9)).containsExactlyInAnyOrder(2, 3)
        Assertions.assertThat(TwoSum().twoSum(intArrayOf(3, 2, 4), 6)).containsExactlyInAnyOrder(1, 2)
        Assertions.assertThat(TwoSum().twoSum(intArrayOf(3, 3), 6)).containsExactlyInAnyOrder(0, 1)
        Assertions.assertThat(TwoSum().twoSum(intArrayOf(0, 1, 0), 0)).containsExactlyInAnyOrder(0, 2)
        Assertions.assertThat(TwoSum().twoSum(intArrayOf(-1, 0, 1, 2), -1)).containsExactlyInAnyOrder(0, 1)
        Assertions.assertThat(TwoSum().twoSum(intArrayOf(-1, 0, 1, 2), 0)).containsExactlyInAnyOrder(0, 2)
    }
}
